<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OptionsDialog</class>
 <widget class="QDialog" name="OptionsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>365</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Options</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/images/cog-solid.png</normaloff>:/images/cog-solid.png</iconset>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>330</y>
     <width>171</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Save</set>
   </property>
  </widget>
  <widget class="QGroupBox" name="gbPath">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>381</width>
     <height>91</height>
    </rect>
   </property>
   <property name="title">
    <string>AVRDUDE location</string>
   </property>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>361</width>
      <height>61</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <property name="spacing">
      <number>2</number>
     </property>
     <item row="0" column="0">
      <widget class="QLineEdit" name="lePath">
       <property name="placeholderText">
        <string>Path to AVRDUDE</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="bSetPath">
       <property name="text">
        <string>Select path</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QPushButton" name="bInitialize">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>381</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Initialize data about parts and programmers</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="gbDefaults">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>140</y>
     <width>381</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Select defaults</string>
   </property>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>361</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="lPart">
       <property name="text">
        <string>Part:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="lProgrammer">
       <property name="text">
        <string>Programmer:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QComboBox" name="cbPart">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="cbProgrammer">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="gbTargets">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>220</y>
     <width>381</width>
     <height>105</height>
    </rect>
   </property>
   <property name="title">
    <string>Targets</string>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>361</width>
      <height>74</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Change name:</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QComboBox" name="cbTargetNames">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="leTargetName">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="acceptDrops">
          <bool>false</bool>
         </property>
         <property name="placeholderText">
          <string>New target name</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bSaveName">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Save name</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QComboBox" name="cbTargetDefault">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chbDefault">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Set as default</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>saveSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>rejectChanges()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSetPath</sender>
   <signal>clicked()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>setPathToAvrDude()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>129</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bInitialize</sender>
   <signal>clicked()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>initializeAvrDudeData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>129</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSaveName</sender>
   <signal>clicked()</signal>
   <receiver>OptionsDialog</receiver>
   <slot>changeTargetName()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbTargetNames</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>leTargetName</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>271</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveSettings()</slot>
  <slot>setPathToAvrDude()</slot>
  <slot>initializeAvrDudeData()</slot>
  <slot>rejectChanges()</slot>
  <slot>setDefaultTarget(int)</slot>
  <slot>changeTargetName()</slot>
 </slots>
</ui>
